sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/state/FollowingViewportState.kt

```diff
@@ -0,0 +1,13 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import com.mapbox.maps.plugin.viewport.experimental.data.FollowingViewportStateOptions
+
+/**
+ * The [ViewportState] that follows user's location.
+ */
+interface FollowingViewportState : ViewportState {
```

Comment:
The interface is introduced so that user can mutate the FollowingViewportStateOptions later, if we make it immutatble,   I think we can get rid of this interface.

Reply:
Capturing from our discussion, we need to return the interface instead of concrete implementation with State factory as a limitation of current architecture, we need to keep this interface to make the options visible to the user and configure on the fly.